tests/unit/rate_limits/test_rate_limit.py

```diff
@@ -157,9 +160,63 @@ def test_determine_if_entity_is_rate_limited_false(self):
             == False
         )
 
+    def test_determine_if_entity_is_rate_limited_error(self, mocker):
+        mock_redis = MagicMock(name="fake_redis")
+        mock_redis.exists.side_effect = RedisError
+        assert (
+            determine_if_entity_is_rate_limited(
+                redis_connection=self.redis_connection,
+                key_name="random_non_existent_key",
+            )
+            == False
+        )
+
     def test_set_entity_to_rate_limited_success(self):
         key_name = "owner_id_123"
         set_entity_to_rate_limited(
             redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=300
         )
         assert self.redis_connection.get(f"rate_limited_entity_{key_name}") is not None
+
+    def test_set_entity_to_rate_limited_error(self, mocker):
+        mock_redis = MagicMock(name="fake_redis")
+        mock_redis.set.side_effect = RedisError
+        key_name = "owner_id_123"
+        # This actually asserts that the error is not raised
+        # Despite the call failing
+        set_entity_to_rate_limited(
+            redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=300
+        )
+        mock_redis.set.assert_not_called()
+
+    def test_set_entity_to_rate_limited_ttl_zero(self):
+        mock_redis = MagicMock(name="fake_redis")
+        mock_redis.set.side_effect = RedisError
+        key_name = "owner_id_123"
+        # This actually asserts that the error is not raised
+        # Despite the call failing
+        set_entity_to_rate_limited(
+            redis_connection=self.redis_connection, key_name=key_name, ttl_seconds=300
```

Comment:
I suppose the TTL should be 0 here.
It's weird the test passed then. It might be passing on the wrong expectation

Reply:
Oh it passed because I accidentally copy pasted a test, changed the title name, and not the implementation. I will fix that